### PR TITLE
Simon & Justines chatbot week 4 

### PR DESCRIPTION
**Pull request by Kyparn**

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -1,47 +1,160 @@
...
-// This means the greeting function will be called one second after the website is loaded.
-setTimeout(greeting, 1000)
+// All the DOM selectors stored as short variables
+const chat = document.getElementById("chat");
+const inputWrapper = document.getElementById("input-wrapper");
+const nameInput = document.getElementById("name-input");
+const form = document.getElementById("name-form");
+
+// const sendBtn = document.getElementById("send-btn")
+const audio = new Audio("drill.mp3");
```
Comment (jiiinCho) [inline code/script.js:8]: great you include Audio!

File: code/script.js [hunk 2]
```diff
@@ -1,47 +1,160 @@
...
+
+// Global variables, if you need any, declared here
+let questionStep = 1;
+
+// Function to display the message on the screen both for bot and user
+const botAnswer = (inputMessage) => {
```
Comment (jiiinCho) [inline code/script.js:14]: it was great that botAnswer and userAnswer is separated, good readability :)

File: code/script.js [hunk 3]
```diff
@@ -1,47 +1,160 @@
...
+  showMessage(inputMessage, "bot");
+};
+const userAnswer = (inputMessage) => {
+  showMessage(inputMessage, "user");
+};
+
+// Question flow functions
+const handleInput = (message) => {
```
Comment (jiiinCho) [inline code/script.js:22]: great you made a flow function!

File: code/script.js [hunk 4]
```diff
@@ -1,47 +1,160 @@
...
+  if (questionStep === 2) {
+    userAnswer(message);
+    setTimeout(() => moreServices(message), 2000);
+  } else if (questionStep === 3) {
+    userAnswer(message);
+    setTimeout(() => priceInformation(message), 1000);
```
Comment (jiiinCho) [inline code/script.js:28]: is there any reason that priceInfomation and moreServices has different timespan? ;)

File: code/script.js [hunk 5]
```diff
@@ -1,47 +1,160 @@
...
+    chat.innerHTML += `
+      <section class="user-msg">
+        <div class="bubble user-bubble">
+          <p>${message}</p>
+        </div>
+        <img src="assets/user.png" alt="User" />  
+      </section>
+    `;
+  } else if (sender === "bot") {
+    console.log("This works");
```
Comment (jiiinCho) [inline code/script.js:46]: maybe this console.log can be deleted when you deployed the website?

File: code/script.js [hunk 6]
```diff
@@ -1,47 +1,160 @@
...
+          <p>${message}</p>
+        </div>
+      </section>
+    `;
+  }
+
+  // Audio
+
+  audio.currentTime = 0;
+  audio.play();
```
Comment (jiiinCho) [inline code/script.js:60]: I see error message in console regarding this line, I am not sure what is the issue though!

File: code/script.js [hunk 7]
```diff
@@ -1,47 +1,160 @@
...
+const serviceOptions = (name) => {
+  questionStep++;
+  botAnswer(`Hi ${name} 👋 What can we help you with today?`);
+  inputWrapper.innerHTML = `
+    <button id="plumbingButton"> Plumbing 🔧 </button>
+    <button id="electricalButton"> Electrical Work 🔌 </button>
+    <button id="paintingButton"> Painting 🖌️ </button>
+    `;
+  document.getElementById("plumbingButton").addEventListener("click", () => {
+    type = "plumbing";
```
Comment (jiiinCho) [inline code/script.js:92]: maybe this 'type' variable is not necessary? Or do you use it somewhere else?

File: code/script.js [hunk 8]
```diff
@@ -1,47 +1,160 @@
...
+
+// 5 th bot message
+function goodBye() {
+  botAnswer(
+    `Thank you for booking ${type} with us, some will contact you soon. `
+  );
+  botAnswer(`Have a lovley day!`);
+  // Below clears the option to click book after once selected.
+  inputWrapper.innerHTML = ``;
+}
```
Comment (jiiinCho) [inline code/script.js:158]: if you'd like to refresh the page when all chat is ended, you can add `window.location.reload(); ` here as our project did!

File: code/script.js (hunk 1 above)
Comment (jiiinCho) [inline code/script.js:8]: it's great you added audio effect in the project!

File: code/script.js (hunk 2 above)
Comment (jiiinCho) [inline code/script.js:14]: it was nice to specify botAnswer and userAnswer, good readability!

File: code/script.js (hunk 3 above)
Comment (jiiinCho) [inline code/script.js:22]: it's good you made question flow function, easy to follow!

File: code/script.js (hunk 5 above)
Comment (jiiinCho) [inline code/script.js:46]: maybe this console.log can be deleted in your deployed website?

File: code/script.js (hunk 7 above)
Comment (jiiinCho) [inline code/script.js:92]: maybe this 'type' variable is not necessary? or do you use it somewhere else?

**Review by jiiinCho — COMMENTED** [review]
It’s great that your project achieved the black level requirements such as sound effects and using questionStep variable to organise the function flow. Just one concern is that there is an error message when I check console in chrome developer’s tool (in your deployed website on netfliy), says ‘Uncaught (in promise) DOMException: play() failed because the user didn't interact with the document first.’ I’m not sure how to fix it, maybe check it later on ! 
It would have been nice if there is a ‘no’ option when booking the service, but it was not a mandatory requirement :)
Great work!

File: code/script.js (hunk 3 above)
Comment (Neaa99) [inline code/script.js:22]: Good and clean structure!

File: code/script.js (hunk 6 above)
Comment (Neaa99) [inline code/script.js:60]: Awesome that you added sounds effect!

File: code/script.js [hunk 9]
```diff
@@ -1,47 +1,160 @@
...
+  event.preventDefault();
+  const name = nameInput.value;
+  userAnswer(`${name}`);
+  // This will keep the input box empty after user clicks send
+
+  nameInput.value = "";
+  setTimeout(() => serviceOptions(name), 1000);
+});
+
+// 2nd Bot message
```
Comment (Neaa99) [inline code/script.js:82]: Very helpfull with the comments that describe what each function do!

File: code/script.js [hunk 10]
```diff
@@ -1,47 +1,160 @@
...
+};
+
+//4th Bot message
+const priceInformation = (service) => {
+  questionStep++;
+  if (service === "emergency repair") {
+    botAnswer(`Perfect! Our starting price for ${service} is SEK 5,000.`);
+  } else {
+    botAnswer(`Perfect! Our starting price for ${service} is SEK 3,000.`);
+  }
```
Comment (Neaa99) [inline code/script.js:140]: Nice touch with the price diffrence and using if/else.

**Review by Neaa99 — COMMENTED** [review]
Great chatbot that works perfectly. The code follows the guidlines, fill the requriments and have a good structure thrught out the code. Great work!

**Pull request closed without merging**